### PR TITLE
Txn fixes (#3971) and arm64 static build

### PR DESCRIPTION
**Pull request by edenhill**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "files.associations": {
+        "string.h": "c",
+        "signal.h": "c"
```
Comment (edenhill) [inline .vscode/settings.json:4]: What's this for? And why only for these two system headers that are not part of librdkafka?
Not really sure this belongs in the repo.

**Comment by edenhill** [comment]
Closing this PR in favour of new future PR.

File: .vscode/settings.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (emasab) [reply .vscode/settings.json:4]: oh, I didn't mean to commit that. I've added it to my local .gitignore only afterwards

File: src/rdkafka_txnmgr.c [hunk 2]
```diff
@@ -3175,3 +3271,20 @@ void rd_kafka_txns_init(rd_kafka_t *rk) {
 
         rd_atomic64_init(&rk->rk_eos.txn_dr_fails, 0);
 }
+
+/**
+ * @brief Complete a pending commit or abort.
+ */
```
[the remark this answers is not in the document]
Comment (emasab) [reply src/rdkafka_txnmgr.c:3277]: I've removed this function

File: src/rdkafka_txnmgr.c [hunk 3]
```diff
@@ -3175,3 +3271,20 @@ void rd_kafka_txns_init(rd_kafka_t *rk) {
...
+void rd_kafka_txns_complete_waiting(rd_kafka_t *rk) {
+        rd_kafka_error_t *error;
+        if ((error = rd_kafka_txn_require_state(
```
[the remark this answers is not in the document]
Comment (emasab) [reply src/rdkafka_txnmgr.c:3280]: I've removed this function

**Pull request closed without merging**